README.md

```diff
@@ -35,10 +35,10 @@ cask_args appdir: "~/Applications", require_sha: true
 
 # 'brew install'
 brew "imagemagick"
-# 'brew install --with-rmtp', 'brew services restart' on version changes
-brew "denji/nginx/nginx-full", args: ["with-rmtp"], restart_service: :changed
-# 'brew install', always 'brew services restart', 'brew link', 'brew unlink mysql' (if it is installed)
-brew "mysql@5.6", restart_service: true, link: true, conflicts_with: ["mysql"]
+# 'brew install --with-rmtp', always 'brew services restart'
+brew "denji/nginx/nginx-full", args: ["with-rmtp"], restart_service: true
+# 'brew install', always 'brew service start' if not running, 'brew services restart' on version changes, 'brew link', 'brew unlink mysql' (if it is installed)
+brew "mysql@5.6", start_service: true, restart_service: :changed, link: true, conflicts_with: ["mysql"]
```

Comment:
```suggestion
brew "mysql@5.6", ensure_started: true, restart_service: :changed, link: true, conflicts_with: ["mysql"]
```

What do you think about this naming? I'm not sure I fully understand if that's what it's doing. I'm also ensure if we're best to add another e.g. `start_service` or add a new state to `restart_service`. Thoughts?

Reply:
Note that this PR uses the existing `start_service` option (it's not a new one!) anticipating a softer landing for an incremental improvement that doesn't break compatibility.

I'd love to see it motivate a DSL change as well, reflecting that common Do What I Mean case of "hey, this Brewfile runs these services, please." In my own exposure to Homebrew bundling scenarios, running the service is so commonly expected that I'd encourage it as a default that needn't be stated at all: the service is started on install and restarted on upgrade. To disable, pass `service: false` to indicate that service management is ignored entirely.

Better yet, move service declarations to a distinct DSL _and_ commandset that echoes the existing `brew services …` suite. Let software deps be software deps only. Declare services with e.g. `services "mysql@5.6", "redis"`. Invoke `brew bundle services …` to act like `brew services … --all` but scoped to the services declared in the Brewfile. Treat `brew bundle install` as implying `brew bundle services start`, with `brew bundle --no-services` to override.

DSL suggestions aside, I think this PR is a sensible step toward declarative services management, demonstrating "Do What I Mean" behavior that such a DSL could be hewn to suit.

Reply:
> Note that this PR uses the existing start_service option (it's not a new one!) anticipating a softer landing for an incremental improvement that doesn't break compatibility.

Yup, it just reads weirdly to me to have to specify both like this.

> the service is started on install and restarted on upgrade. T

I don't agree with this because many services are installed e.g. as dependencies but never need to be run.

> I'd love to see it motivate a DSL change as well, reflecting that common Do What I Mean case of "hey, this Brewfile runs these services, please."

I've always used and had users of my `Brewfile`s use `restart_service` for this and this PR is the first I've heard complaints about the service being unnecessarily restarted.

> DSL suggestions aside, I think this PR is a sensible step toward declarative services management, demonstrating "Do What I Mean" behavior that such a DSL could be hewn to suit.

As-is: I'm afraid I disagree. With some iteration/naming changes: quite possibly 👍🏻

Reply:
* [Reverted](https://github.com/Homebrew/homebrew-bundle/pull/1115/commits/f7b441d64a103d18ccbc74d2458118fb058d261b) the dumper changes that introduced confusing dual usage of `start_service` alongside `restart_service`.
* [Proposed](https://github.com/Homebrew/homebrew-bundle/pull/1115/commits/018bf7130dc463f4da0644b67d9a3bbd852e683d) having `restart_service: :changed` also start services that aren't running.

Reply:
> In this usage, I have yet to encounter a declared dependency that does not need to run.

This definitely happens. Some software that provides a service is needed as a build-time dependency of non-Homebrew software.

> Apologies; this PR is not a complaint. I appreciate and extensively rely on Homebrew Bundle.

No worries! Didn't take it as one ❤️ 

> That'd achieve the same outcome, that install will start a service that states restart_service: :changed and check will pass afterward.

That would work for me 👍🏻

Reply:
> > That'd achieve the same outcome, that install will start a service that states restart_service: :changed and check will pass afterward.
> That would work for me 👍🏻

(This is done, btw! PR now boils down to `restart_service: :changed` implying `start_service: true`.)